### PR TITLE
fix: $nuxt is missing from the Vue instance

### PR DESCRIPTION
**Pull request by wangxingkai**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

fix: $nuxt is missing from the Vue instance when a customised globalName has been specified

[`packages/vue-app/template/components/nuxt-link.client.js`](https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-app/template/components/nuxt-link.client.js#L69) uses `this.$nuxt.isOffline` which gives an error

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/vue-app/template/App.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ export default {
   },
   created() {
     // Add this.$nuxt in child instances
-    Vue.prototype.<%= globals.nuxt %> = this
+    Vue.prototype.<%= globals.nuxt %> = <%= (globals.nuxt !== '$nuxt' ? 'Vue.prototype.$nuxt = ' : '') %>this
```
Comment (clarkdo) [inline packages/vue-app/template/App.js]: I think `globals` is for hiding `$nuxt` actually, so maybe change `nuxt-link.client.js` to use dynamic `globals.nuxt` is better 😃

**Comment by wangxingkai** [comment]
Change 2 places that using hardcoded `this.$nuxt` to `<%= globals.nuxt %>`.

**Comment by wangxingkai** [comment]
@pi0 Hey Pooya, would you please also have a look at this PR. The error keeps showing in the console, really want to have it fixed. Thanks.

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
Thanks, @wangxingkai  for your help and sorry for late merge.

**Pull request merged**